.github/workflows/web_app_deploy.yaml

```diff
@@ -0,0 +1,184 @@
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        description: The name of the workspace to create the artifact for.
+        type: string
+        required: true
+      target_name:
+        description: "(Optional) The name of the target to create artifact for (default: main)"
+        type: string
+        required: false
+        default: main
+      environment:
+        description: Environment where the artifact will be deployed.
+        type: string
+        required: true
+      resource_group_name:
+        description: Web App resource group name.
+        type: string
+        required: true
+      web_app_name:
+        description: Web App name.
+        type: string
+        required: true
+      health_check_path:
+        description: The health probe path exposed by the Web App.
+        type: string
+        required: true
+      use_staging_slot:
+        description: True if artifact should be deployed to staging slot
+        type: boolean
+        required: false
+        default: true
+      use_private_agent:
+        description: Use a private agent to deploy the built artifact.
+        type: boolean
+        required: false
+        default: true
+
+concurrency:
+  group: ${{ github.workflow }}-cd
+  cancel-in-progress: true
+
+env:
+  BUNDLE_NAME: bundle
+
+jobs:
+  build:
+    name: Build Artifact
+    runs-on: ubuntu-22-04
+    env:
+      WORKSPACE: ${{ inputs.workspace_name }}
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        name: Checkout
+
+      - name: Prune
+        run: npx turbo@1.13.3 prune --scope ${{ env.WORKSPACE }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          node-version-file: ".node-version"
+          cache: "yarn"
+          cache-dependency-path: "./out/yarn.lock"
+
+      - name: Install dependencies
+        run: yarn install --immutable
+        working-directory: ./out
+
+      - name: Build
+        run: yarn build
+        working-directory: ./out
+
+      - name: Build the artifact
+        id: make-artifact
+        env:
+          YARN_NODE_LINKER: node-modules
```

Comment:
```suggestion
         # we fall back to node-moules, even in case pnp is configured, in order to avoid bundling dependendencies
          YARN_NODE_LINKER: node-modules
```